src/modalities/models/model_factory.py

```diff
@@ -266,6 +266,9 @@ def get_parent_module_and_child_name(child_module: nn.Module, model: nn.Module)
             module_class = get_module_class_from_name(model, name)
             if module_class is not None:
                 block_types.append(module_class)
+            if block_names and not block_types:
```

Comment:
What about raising an exception the moment a  `module_class` cannot be found, i.e., 
```suggestion
            else:
```